### PR TITLE
Added non_blocking capability to convert_tensor, prepare_batch, create_supervised_trainer

### PR DESCRIPTION
**Pull request by anmolsjoshi**
Pull request for this [issue](https://github.com/pytorch/ignite/issues/231)

Adds **kwargs to prepare_batch, create_supervised_trainer, convert_tensor

convert_tensor has a check for non_blocking in kwargs, otherwise assigns False.

### CODE REVIEW

**Comment by vfdev-5** [comment]
There tests that are missing. That's why CI fails.

**Comment by vfdev-5** [comment]
@anmolsjoshi may I ask you why you think it is better to provide kwargs instead of just `non_blocking` attribute ?

**Comment by anmolsjoshi** [comment]
@vfdev-5 thanks for your comments. I misunderstood the discussion on the issue about using kwargs, reading it now it was meant as more of a general statement. I have incorporated that comment. I had initially written it without kwargs, sorry! 

Is there a method to test non_blocking?

**Comment by vfdev-5** [comment]
@anmolsjoshi  no problem, I just wonder what else can we pass in kwargs to the method `.to()` 
```
to(device=None, dtype=None, non_blocking=False) → Tensor
```
[ref](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.to)
We are already passing device, about dtype I'm not sure and remains `non_blocking` :)

> Is there a method to test non_blocking?

actually you just need to have tests that passes all cases of 
```python
    if 'non_blocking' in kwargs:
        non_blocking = kwargs.pop('non_blocking')
    else:
        non_blocking = False
```
so just one test with `non_blocking=True` and another with `non_blocking=None` in your current implementation.

**Comment by anmolsjoshi** [comment]
@vfdev-5 given that only other parameters we pass to .to() is device, dtype and non_blocking. I'm going to change the code to pass non_blocking directly as previously suggested.

File: ignite/engine/__init__.py [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
-def _prepare_batch(batch, device=None):
+def _prepare_batch(batch, device=None, non_blocking=False):
     x, y = batch
-    return convert_tensor(x, device=device), convert_tensor(y, device=device)
+    return (convert_tensor(x, device=device, non_blocking=non_blocking),
+            convert_tensor(y, device=device, non_blocking=non_blocking))
 
 
-def create_supervised_trainer(model, optimizer, loss_fn, device=None):
+def create_supervised_trainer(model, optimizer, loss_fn, device=None, non_blocking=False):
```
Comment (alykhantejani) [inline ignite/engine/__init__.py]: Can we add `non_blocking` to the docstring

Comment (anmolsjoshi) [reply ignite/engine/__init__.py]: Incorporated and committed.

**Comment by alykhantejani** [comment]
I guess this build fail is down to visdom on travis @vfdev-5 ?

**Comment by vfdev-5** [comment]
@alykhantejani yes you are right. I added an issue on this #241

**Comment by vfdev-5** [comment]
Sorry for the delay with failed CI. As it is fixed now, we can merge the PRs.
@anmolsjoshi thanks !

**Pull request merged**